cli/cmd/install.go

```diff
@@ -834,61 +850,51 @@ func (options *installOptions) proxyConfig() *pb.Proxy {
 	}
 }
 
-// exitIfClusterExists checks the kubernetes API to determine
-// whether a config exists and exits if it does exist or if an error is
-// encountered.
-//
-// This bypasses the public API so that public API errors cannot cause us to
-// misdiagnose a controller error to indicate that no control plane exists.
-func exitIfClusterExists() {
-	k, err := k8s.NewAPI(kubeconfigPath, kubeContext, 0)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
-		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
-		os.Exit(1)
+func errIfGlobalResourcesExist() error {
+	checks := []healthcheck.CategoryID{
+		healthcheck.KubernetesAPIChecks,
+		healthcheck.LinkerdPreInstallGlobalResourcesChecks,
 	}
 
-	c := k.CoreV1().ConfigMaps(controlPlaneNamespace)
-	if _, err = c.Get(k8s.ConfigConfigMapName, metav1.GetOptions{}); err != nil {
-		if kerrors.IsNotFound(err) {
-			return
+	hc := healthcheck.NewHealthChecker(checks, &healthcheck.Options{
+		ControlPlaneNamespace: controlPlaneNamespace,
+		KubeConfig:            kubeconfigPath,
+	})
+
+	errMsgs := ""
+	hc.RunChecks(func(result *healthcheck.CheckResult) {
+		if result.Err != nil {
+			errMsgs += fmt.Sprintf("%s", result.Err)
```

Comment:
i think this will just concatenate error strings together?

consider:
```go
errMsgs := []string{}
hc.RunChecks(func(result *healthcheck.CheckResult) {
	if result.Err != nil {
		errMsgs = append(errMsgs, result.Err.Error())
	}
})

if len(errMsgs) > 0 {
	return errors.New(strings.Join(errMsgs, ", "))
}
```